utils/writeable/src/lib.rs

```diff
@@ -158,7 +158,7 @@ pub trait PartsWrite: fmt::Write {
 }
 
 /// `Writeable` is an alternative to `std::fmt::Display` with the addition of a length function.
-pub trait Writeable {
+pub trait Writeable: fmt::Display {
```

Comment:
Issue: This is not a good use of a trait bound. Trait bounds should be used only if the inner trait is required as a dependency of the defined trait.

Reply:
Yeah we want a lint that requires `Display` is always implemented, but we can't do that.

Reply:
Yeah I would agree that we should perhaps require Display in a bunch of places, but not enforce it this way

Especially since writeable is a util crate (and in general we should try not to make our util crates too ICU4X-specific)


The normal way to do this is for us to find where we have Writeable bounds and add +Display to them if necessary, but I'm not sure that works well either.

I think we can just commit to best-effort making everything Display.

Reply:
The point here is that all our output types should be `Display`, which we cannot enforce by trait bounds, we just have to enforce. My thinking was that anything that's `Writeable` is also `Display` so we that there's compatibility with `core::fmt`.

Reply:
Yeah I think we should require it on our FormattedFoos, but Writeable shouldn't have to imply it, especially since we're changing a util crate to enforce something that ICU4X specifically wants (but perhaps not the general users)